libdevice/imf_utils/saturatef.cpp

```diff
@@ -0,0 +1,16 @@
+//==------ saturatef.cpp - fallback implementation of __imf_saturatef ------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "../device_imf.hpp"
+
+#ifdef __LIBDEVICE_IMF_ENABLED__
+
+DEVICE_EXTERN_C_INLINE
+float __devicelib_imf_saturatef(float x) { return __fclamp(x, .0f, 1.f); }
```

Comment:
Numerics team owns a number of trivial math functions like __devicelib_imf_copysign, __devicelib_imf_fabs, __devicelib_imf_rsqrt which can be either mapped to just one device independent builtin or implemented by a couple of simple operations - these functions are better to be inlined from header file than called from binary. We would like to gather them all under the ./imf folder which our team owns, promotes and supports. All trivial functions may combined within  just one file. Could you please add such header inside ./imf, say, ./imf/imf_inline.hpp, with one function example (floor?)? Thanks!

Reply:
Hi, @akolesov-intel 
If we want to inline those trivial functions in header, then the header should be exposed to users, users will include it in their  source code. And in the header file, we need to include:
1. declarations for all imf functions provided by us.
2. Some trivial functions inline definition.

Reply:
I am fine if these functions are going to be inlined as IR, not as sources in header. I am just suggesting to gather them all into one file, not to spread by many separate tiny sources. I did that in my package through imf_inline.cpp source (can be named differently).

Reply:
Hi, @akolesov-intel 
I updated the patch to integrate all "trivial inline" functions, those functions are inlined as IR.
If we inline in LLVM IR level, we need also split imf_inline.cpp into imf_inline_fp32.cpp and imf_inline_fp64.cpp, I did this in the latest patch and basic tests for fp32, fp64, fp16 passed. Could you help review?
Thanks very much.

Reply:
Thank you, @jinge90 , the split looks good to me.